### ISSUE TITLE
[Accessibility]: Page content changes on navigation without announcement

### ISSUE THREAD

**Issue by efritz** [sourcegraph/sourcegraph#35181]
### Audit type

Screen reader navigation

### User journey audit issue

#33516, #33518, #33513, #33515

### Problem description

In site-admin **and** the repository-specific code intelligence settings page, the side bar navigation does not indicate that anything on the page has changed. Going to the next header (in all the cases that I've tried) will jump up to the top the new content.

<img width="1164" alt="Screen Shot 2022-05-09 at 4 17 19 PM" src="https://user-images.githubusercontent.com/103087/167500809-be5519ea-0695-4ddf-bc2b-ae2bd71794ac.png">

### Expected behavior

I would further expect that when I click on a navigation link in the sidebar that my focus will be on the new content immediately. We should re-focus on the top of the new content and announce it via screen reader.

### Assigning labels

- [X] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [X] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [X] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

I'm unsure who should work on this issue or if it needs more clarification. I will let the Frontend Platform team triage this.

**Comment by umpox** [comment]
Fixed through https://github.com/sourcegraph/sourcegraph/pull/35413. This page has a correct h2 heading, meaning screen readers can quickly jump to the content.
